src/gui/torrentcreatordlg.h

```diff
@@ -28,52 +28,63 @@
  * Contact : chris@qbittorrent.org
  */
 
-#ifndef CREATE_TORRENT_IMP_H
-#define CREATE_TORRENT_IMP_H
+#ifndef TORRENTCREATORDLG_H
+#define TORRENTCREATORDLG_H
 
-#include "ui_torrentcreatordlg.h"
+#include <QDialog>
+
+#include "base/settingsstorage.h"
+
+namespace Ui
+{
+    class TorrentCreatorDlg;
+}
 
 namespace BitTorrent
 {
     class TorrentCreatorThread;
 }
 
-class TorrentCreatorDlg : public QDialog, private Ui::createTorrentDialog
+class TorrentCreatorDlg: public QDialog
 {
     Q_OBJECT
 
 public:
-    TorrentCreatorDlg(QWidget *parent = 0);
+    TorrentCreatorDlg(QWidget *parent = 0, const QString &defaultPath = QString());
     ~TorrentCreatorDlg();
-    int getPieceSize() const;
+    void updateInputPath(const QString &path);
 
-public slots:
+private slots:
     void updateProgressBar(int progress);
-    void on_cancelButton_clicked();
-
-protected slots:
-    void on_createButton_clicked();
-    void on_addFile_button_clicked();
-    void on_addFolder_button_clicked();
-    void handleCreationFailure(QString msg);
-    void handleCreationSuccess(QString path, QString branch_path);
-    void setInteractionEnabled(bool enabled);
-    void showProgressBar(bool show);
-    void on_checkAutoPieceSize_clicked(bool checked);
-    void updateOptimalPieceSize();
-    void saveTrackerList();
-    void loadTrackerList();
-
-protected:
-    void closeEvent(QCloseEvent *event);
+    void onCreateButtonClicked();
+    void onAddFileButtonClicked();
+    void onAddFolderButtonClicked();
+    void handleCreationFailure(const QString &msg);
+    void handleCreationSuccess(const QString &path, const QString &branchPath);
 
 private:
+    void dropEvent(QDropEvent *event) override;
+    void dragEnterEvent(QDragEnterEvent *event) override;
+
     void saveSettings();
     void loadSettings();
+    int getPieceSize() const;
+    void setInteractionEnabled(bool enabled);
 
-private:
+    Ui::TorrentCreatorDlg *ui;
     BitTorrent::TorrentCreatorThread *m_creatorThread;
-    QList<int> m_pieceSizes;
+
+    // settings
+    StorableValue<QSize> storeDialogSize;
```

Comment:
Don't forget about the coding style! We use `m_` prefix for private fields.
I have used it in my examples, but you didn't see it... :(